File: backend/src/main/java/com/ihecride/models/Evaluation.java
package com.ihecride.models;

import java.sql.Timestamp;

public class Evaluation {
    private int id;
    private int trajetId;
    private int evaluateurId;
    private int evalueId;
    private String evaluateurNom;
    private int note;
    private String commentaire;
    private Timestamp date;

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public int getTrajetId() { return trajetId; }
    public void setTrajetId(int trajetId) { this.trajetId = trajetId; }
    public int getEvaluateurId() { return evaluateurId; }
    public void setEvaluateurId(int evaluateurId) { this.evaluateurId = evaluateurId; }
    public int getEvalueId() { return evalueId; }
    public void setEvalueId(int evalueId) { this.evalueId = evalueId; }
    public String getEvaluateurNom() { return evaluateurNom; }
    public void setEvaluateurNom(String evaluateurNom) { this.evaluateurNom = evaluateurNom; }
    public int getNote() { return note; }
    public void setNote(int note) { this.note = note; }
    public String getCommentaire() { return commentaire; }
    public void setCommentaire(String commentaire) { this.commentaire = commentaire; }
    public Timestamp getDate() { return date; }
    public void setDate(Timestamp date) { this.date = date; }
}

File: backend-php/models/Evaluation.php
<?php
/**
 * Entité Evaluation : note + commentaire après un trajet.
 * Équivalent PHP de com.ihecride.models.Evaluation.
 */

class Evaluation implements JsonSerializable {

    public ?int    $id = null;
    public ?int    $trajetId = null;
    public ?int    $evaluateurId = null;
    public ?int    $evalueId = null;
    public ?string $evaluateurNom = null;
    public ?int    $note = null;
    public ?string $commentaire = null;
    public ?string $date = null;

    public function jsonSerialize(): array {
        return [
            'id'             => $this->id,
            'trajetId'       => $this->trajetId,
            'evaluateurId'   => $this->evaluateurId,
            'evalueId'       => $this->evalueId,
            'evaluateurNom'  => $this->evaluateurNom,
            'note'           => $this->note,
            'commentaire'    => $this->commentaire,
            'date'           => $this->date,
        ];
    }
}
